nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_constructor.rs

```diff
@@ -112,12 +116,35 @@ impl StringConstructor {
         Ok(s.into_value())
     }
 
+    /// ### [22.1.2.1 String.fromCharCode ( ...`codeUnits` )](https://262.ecma-international.org/15.0/index.html#sec-string.fromcharcode)
+    ///
+    /// > This function may be called with any number of arguments which form
+    /// the rest parameter `codeUnits`.
     fn from_char_code(
-        _agent: &mut Agent,
+        agent: &mut Agent,
         _this_value: Value,
-        _arguments: ArgumentsList,
+        code_units: ArgumentsList,
     ) -> JsResult<Value> {
-        todo!();
+        // 1. Let result be the empty String.
+        if code_units.is_empty() {
+            return Ok(String::EMPTY_STRING.into_value());
+        }
+        let mut result: Vec<u8> = Vec::with_capacity(code_units.len() * 2);
+
+        // 2. For each element next of codeUnits, do
+        //   a. Let nextCU be the code unit whose numeric value is ℝ(? ToUint16(next)).
+        //   b. Set result to the string-concatenation of result and nextCU.
+        for next in code_units.iter() {
+            let code_unit = next.to_uint16(agent)?;
+            result.extend_from_slice(&code_unit.to_ne_bytes());
+        }
+
+        // 3. Return result.
+        let result_str = str::from_utf8(result.as_slice()).unwrap();
```

Comment:
issue: what is the correct behavior here? Do we have a way of casting structs implementing `std::error::Error` to `JsError`?

Reply:
We indeed do not have that, and at the engine level we also shouldn't really need one. Any errors that the engine produces should be just plain JS `TypeError`, `SyntaxError` etc. A runtime would definitely want to have something like that, though.

The correct thing here would probably be to create a `Wtf8Buf`, though that may be a bit trickier.